processing/src/test/java/org/apache/druid/math/expr/VectorExprResultConsistencyTest.java

```diff
@@ -348,6 +349,28 @@ public void testSymmetricalBivariateFunctions()
     testFunctions(types, templates, functions);
   }
 
+  @Test
+  public void testIfFunction()
+  {
```

Comment:
Would be good to include some tests to validate that lazy evaluation works, such as `if(l1 % 2 == 0, l2 / (l1 % 2), -1)`. That would throw a division-by-zero exception if lazy evaluation wasn't working.

Reply:
added test, though your comment has the then/else swapped because ^ is always divide by zero 😅